sbc_translation/models/correspondence.py

```diff
@@ -22,22 +34,114 @@ class Correspondence(models.Model):
     ##########################################################################
     #                              ORM METHODS                               #
     ##########################################################################
+
     @api.model
     def create(self, vals):
-        """ Create a message for sending the CommKit only if the preferred
-            language is the same as the letter's language else send the letter
-            on the local translate plaforme.
-            """
-
+        """ Create a message for sending the CommKit after be translated on
+             the local translate plaforme.
+        """
         sponsorship = self.env['recurring.contract'].browse(
             vals['sponsorship_id'])
-        if sponsorship.reading_language.id == vals['original_language_id']:
-            letter = super(Correspondence, self).create(vals)
-        else:
+
+        original_lang = self.env['res.lang.compassion'].browse(
+            vals['original_language_id'])
+
+        if original_lang.translatable and original_lang.id not in sponsorship\
+                .child_id.project_id.country_id.spoken_lang_ids.mapped('id'):
             letter = super(Correspondence, self.with_context(
                 no_comm_kit=True)).create(vals)
-            self.alternative()
+            letter.send_local_translate()
+        else:
+            letter = super(Correspondence, self).create(vals)
+
         return letter
 
-    def alternative(self):
-        pass
+    def send_local_translate(self):
+        # Insert the letter in the mysql data base
+        tc = translate_connector.TranslateConnect()
+        # Send letter to local translate platform
+        text_id = tc.upsert_text(self.sponsorship_id, self)
+        translation_id = tc.upsert_translation(text_id, self)
+        tc.upsert_translation_status(translation_id)
```

Comment:
I would write a `upsert_letter` method inside TranslateConnect that call those 3 methods, so that you don't have to pass the parameters each time, like you did for the `remove_letter` method.